tests/integration/pilot/mcp/synthetic_serviceentry_test.go

```diff
@@ -0,0 +1,270 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mcp
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+
+	"istio.io/istio/galley/pkg/metadata"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/pilot"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/util/structpath"
+	"istio.io/istio/tests/integration/pilot/mcp/testdata"
+)
+
+var (
+	ist istio.Instance
+	p   pilot.Instance
+	g   galley.Instance
+)
+
+func TestMain(m *testing.M) {
+	framework.NewSuite("mcp", m).
+		RequireEnvironment(environment.Native).
+		Label(label.CustomSetup).
```

Comment:
I don't see a custom setup here.  I think you could probably run this as part of the top-level pilot tests (i.e. move the file up one level).